openandroidinstaller/tooling.py

```diff
@@ -574,24 +576,49 @@ def heimdall_flash_recovery(bin_path: Path, recovery: str) -> TerminalResponse:
         yield line
 
 
-def search_device(platform: str, bin_path: Path) -> Optional[str]:
+@dataclass(frozen=True)
+class SearchResult:
+    """Result of the device search.
+
+    Attributes:
+        device_code: The device code of the connected device.
+        msg: Message describing the result.
+    """
+
+    device_code: str = None
+    msg: str = None
+
+
+def search_device(platform: str, bin_path: Path) -> SearchResult:
     """Search for a connected device."""
     logger.info(f"Search devices on {platform} with {bin_path}...")
     try:
         # read device code
         if platform in ("linux", "darwin"):
+            # check if grep is installed and find the right path
+            try:
+                grep_command = check_output(["which", "grep"]).decode().strip()
+            except CalledProcessError:
+                logger.error(
+                    "Failed to detect a device. Please make sure `grep` it is installed."
```

Comment:
"Please make sure `grep` is installed" sounds better, isn't it?

Reply:
Thanks for having a look! Sounds better indeed! :)